tools/topology/m4/muxdemux.m4

```diff
@@ -2,6 +2,24 @@ divert(-1)
 
 dnl Define macro for demux widget
 
+define(comp_type, 0)
+define(sof_magic, 0x00464F53)
+define(stream_data_size, 16)
+define(mux_config_size, 8)
+
+define(ROUTE_MATRIX, `PRINT_BYTES_4($1),PRINT_BYTE($2),PRINT_BYTE($3),PRINT_BYTE($4),PRINT_BYTE($5),PRINT_BYTE($6),PRINT_BYTE($7),PRINT_BYTE($8),PRINT_BYTE($9),PRINT_BYTE($10),PRINT_BYTE(0),PRINT_BYTE(0),PRINT_BYTE(0)')dnl
+
+define(`MUXDEMUX_CONFIG',
+`SectionData.STR($1) {'
+`bytes "'`PRINT_BYTES_4(sof_magic),PRINT_BYTES_4(comp_type),'
+       `PRINT_BYTES_4(eval(mux_config_size + (stream_data_size * $4))),PRINT_BYTES_4(SOF_ABI_VERSION),'
+       `PRINT_BYTES_4(0),PRINT_BYTES_4(0),'
+       `PRINT_BYTES_4(0),PRINT_BYTES_4(0),'
+       `PRINT_BYTES_2($2),PRINT_BYTES_2($3),PRINT_BYTES_2($4),PRINT_BYTES_2(0),'
+       `$5'`"'
+       `}'
```

Comment:
same, is there a description of the representation?

Reply:
ok, I tried to add the reference to the actual struct. and comment that reserved members are zeroed.